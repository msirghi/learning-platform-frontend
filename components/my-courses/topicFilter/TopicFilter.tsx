import React from 'react';
import { TextField } from '@material-ui/core';
import styles from './TopicFilter.module.scss';
import { useWindowSize } from '../../common/hooks/useWindowResize';
import { useTranslation } from '../../../i18n';

type Props = {
  value: string;
  onChange: (val: string) => void;
};

export const TopicFilter: React.FC<Props> = ({ value, onChange }) => {
  const [width] = useWindowSize();
  const { t } = useTranslation();

  return (
    <div className={styles.searchFieldWrapper}>
      <div className={styles.field}>
        <TextField
          label={t('courses:findTopic')}
          fullWidth={width < 768}
          value={value}
          onChange={(e) => onChange(e.target.value)}
        />
      </div>
    </div>
  );
};
