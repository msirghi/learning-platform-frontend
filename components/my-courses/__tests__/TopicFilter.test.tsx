import { TextField } from '@material-ui/core';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { TopicFilter } from '../TopicFilter';

describe('TopicFilter component', () => {
  beforeAll(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<TopicFilter value='val' onChange={jest.fn()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have proper input value', () => {
    const wrapper = shallow(<TopicFilter value='val' onChange={jest.fn()} />);
    const input = wrapper.find(TextField);

    expect(input).toHaveLength(1);
    expect(input.props().value).toEqual('val');
  });

  it('should have input that is not full width on desktop', () => {
    const initWidth = global.innerWidth;
    global.innerWidth = 1024;
    global.dispatchEvent(new Event('resize'));

    const wrapper = mount(<TopicFilter value='val' onChange={jest.fn()} />);
    const input = wrapper.find(TextField);

    wrapper.update();
    expect(input).toHaveLength(1);
    expect(input.props().fullWidth).toBeFalsy();
    global.innerWidth = initWidth;
    global.dispatchEvent(new Event('resize'));
  });

  it('should have input that is full width on mobile', () => {
    const initWidth = global.innerWidth;
    global.innerWidth = 500;
    global.dispatchEvent(new Event('resize'));
    const wrapper = shallow(<TopicFilter value='val' onChange={jest.fn()} />);
    const input = wrapper.find(TextField);

    expect(input).toHaveLength(1);
    expect(input.props().fullWidth).toBeTruthy();
    global.innerWidth = initWidth;
    global.dispatchEvent(new Event('resize'));
  });

  it('should call onChange prop on input change', () => {
    const spy = jest.fn((data) => data);
    const wrapper = shallow(<TopicFilter value='val' onChange={spy} />);
    const input = wrapper.find(TextField);

    expect(input).toHaveLength(1);
    input.props().onChange({ target: { value: 'new value' } });
    expect(spy).toBeCalled();
    expect(spy).toBeCalledWith('new value');
  });
});
