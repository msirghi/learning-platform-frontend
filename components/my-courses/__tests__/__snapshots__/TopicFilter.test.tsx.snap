// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicFilter component should match the snapshot 1`] = `
<div
  className="searchFieldWrapper"
>
  <div
    className="field"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="courses:findTopic"
      onChange={[Function]}
      value="val"
    />
  </div>
</div>
`;
